### ISSUE TITLE
Ability to mark a session as "NSFW", and option to hide it from Active Sessions view

### ISSUE THREAD

**Issue by NukiRaccoon** [Yellow-Dog-Man/Resonite-Issues#1022]
### Is your feature request related to a problem? Please describe.

There have been multiple issues of people creating sessions, public and semi-private, which contain content which is considered "NSFW".  This is NOT simply 18+ or sexual content, but simply content you would not want to explain being on your computer screen in a professional or family setting where your coworkers, parents, or children might see it.

This includes both public worlds which contain content which simply "makes people uncomfortable", and semi-private worlds which require external verification to see.

Currently, the Preview puts whatever is in these sessions on full display, which has caused issues.  I have personally had to explain to a friend what a particular 18+ private session was.

### Describe the solution you'd like

In the session options, add a checkbox which marks the session as having "Sensitive Content" or being "NSFW" or something.

Add this as a flag for the session viewer.

In the Session Viewer, add an option to show or hide these sessions at the top of the interface.

Only show this option to accounts which are more than 24 hours old, so they can only see these sessions if added manually.

This should all also function on the World level, for worlds which contain "sensitive content".

### Describe alternatives you've considered

Currently, the alternative is to use advanced knowledge of the workings of Resonite to 1) Change the preview to something other than whatever is on screen, 2) have some sort of "accept or deny" popup which only appears AFTER the world is loaded, and 3) I guess change the name of the session to something with a CW like on Mastodon....  And even then, new users and users who are sensitive and will complain can STILL see and join.

### Additional Context

This IS related to current drama, but I have several experiences which are unrelated where this would be good: I would like to use it on a world I am currently working on in Blender.

**Comment by Frooxius** [comment]
You can already mark sessions as "Hidden", which hides them from the view and will prevent previews from being shown to users.

Marking it explicitly as NSFW has larger implications and probably not something we will add in the immediate future. There's a number of issues to be resolved with that:
- We need to figure out the legalities of this, because this might put some responsibility on our end (such as age verification)
- Policy on what's NSFW and what isn't - the line can be difficult to draw for some things and we'll need to moderate the sessions in some way, so this might not even resolve the issue that is currently being tackled - if what you consider uncomfortable is not considered as NSFW, this option will not help and is not the right solution for this
- What would this actually do on top of hiding the session (which you can already do)? Would it prevent underage users from joining? What standard would we need to use to ensure of that?

There is a lot of issues that we need to work through if we decide we'd want such option to happen. 

I'd just recommend using "Hidden" session for now if you don't want it to be displayed.

**Comment by Azavit** [comment]
I could see a "Sensitive Content" Tag (rather than NSFW due to the situations mentioned above) that just blurs the visual of the world until you expand it/hover over it and maybe even gives a little list of Content Warnings.

This isn't "hiding" it per say as nothing about the world is wrong to show to anyone of age but would allow things like "Arachnid World" that is filled with a bunch of spiders from triggering those with arachnophobia.

**Comment by Frooxius** [comment]
It would be a bit different from NSFW and wouldn't address the original issue, because we likely wouldn't allow you to use this for NSFW, since doing that would still need those problems tackled.

**Comment by Azavit** [comment]
> This is NOT simply 18+ or sexual content, but simply content you would not want to explain being on your computer screen in a professional or family setting where your coworkers, parents, or children might see it.

It was explained in the original post that it was "Not Safe for Work" (Content that may not be appropriate for work environments) and that it was explicitly not talking about content that was 18+ nor sexual content.  They also suggested the term "Sensitive Content" themselves which is why I made my suggestion.

**Comment by Frooxius** [comment]
The problem is that people are likely to use it that way, which means we still need to consider the case when they do and what do we do if it ends up being misused.

**Comment by FlameSoulis** [comment]
I mean, to my knowledge, many websites, like Twitter, don't seem to do any major verifications (despite a push for absolute verification, which is not quite the same scope). I'm guessing where realtime user engagement is involved, things change?

I do like the sensitive content concept, as it feels like a safer-in-between action, and could be useful. This could also allow for more creativity, such as horror maps that replicate modern movies (basically bankrupting ketchup factories, the special effects department).

**Comment by Geenz** [comment]
Two things here, both are incredibly complex and are not likely to be implemented for a bit:
* Content tagging
* Age gating

Note that I'm separating _verification_ from gating.

There's a lot of complexity with doing something like this - and right now we haven't started work on either of these things.

**Comment by NukiRaccoon** [comment]
> Two things here, both are incredibly complex and are not likely to be implemented for a bit:
> 
> * Content tagging
> * Age gating
> 
> Note that I'm separating _verification_ from gating.
> 
> There's a lot of complexity with doing something like this - and right now we haven't started work on either of these things.

What do you think about adding a simple text-block which users have to agree to the contents of before joining?  This could be for both session rules and for content tagging, at least in the short-term.  It wouldn't be searchable, but it would be a partial implementation of what's being asked for here.

That would be this issue...
https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/1020#issuecomment-1891117588

**Comment by Geenz** [comment]
It is something we’ve been contemplating- however it is not a solution to this.

**Comment by The-furf-of-July** [comment]
My primary reason for supporting this as a feature is, an issue I've had in the past is how isolated sessions can get when such content is present. 
In my experience, socialisation in explicit settings or topics is made extremely difficult by the requirement for the world to remain hidden and unjoinable until said content is gone. It makes meetups more difficult, and finding those communities near impossible.
(Unless there have been policy changes that addressed this that I am not aware of)


A potential alternative solution would be groups or contact tagging/filtering, so that users could verify on their own which contacts they want in certain contexts, and open the world to people they are personally comfortable with. This would also take the burden of age verification and put it on the host.

**Comment by gentlecolts** [comment]
Something I'd like to emphasize from my prior ticket is that 18+ doesn't necessarily have to mean nsfw, I'd personally prefer emphasis to remain on age gating, and let the session hosts decide whether or not the environment should be nsfw.

**Comment by NukiRaccoon** [comment]
Yeah the point here isn't to have content that would otherwise break the current rules for public sessions, the point here is simply adding a gate.  To add some context, I believe this was one of the feature requests I made back during a period where someone was upset about some of the content in public worlds, and I was trying to "be the change" by actually requesting some of the features that came up in that Discord thread.
